src/renderers/shared/fiber/ReactFiberBeginWork.js

```diff
@@ -277,6 +277,11 @@ module.exports = function<T, P, I, TI, PI, C, CX, PL>(
     markRef(current, workInProgress);
 
     if (!shouldUpdate) {
+      // Context providers should defer to sCU for rendering
+      if (hasContext) {
+        invalidateContextProvider(workInProgress, false);
```

Comment:
But as long as it's in the codebase, I think we should keep the implementations in sync.